### PR TITLE
nixos/tautulli: add option to open firewall

### PR DESCRIPTION
**Pull request by rhoriguchi**
Naming of old branch was a pain, so I merged the Pr #152159 with this one. 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by csingley — APPROVED** [review]
Thanks for doing this!  You might get some pushback on opening firewall ports here though.  I initially had this in the derivation and was talked into removing them.

**Comment by rhoriguchi** [comment]
> Thanks for doing this! You might get some pushback on opening firewall ports here though. I initially had this in the derivation and was talked into removing them.

I can find a 145 [openFirewall](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=openFirewall) options. Also in my opinion this is much cleaner than having `networking.firewall.allowedTCPPorts` with a lot of different ports, so you always have to check what I still needed. From a user perspective, this is a lot simpler.

File: nixos/modules/services/misc/tautulli.nix [hunk 1]
```diff
@@ -32,6 +32,12 @@ in
         description = lib.mdDoc "TCP port where Tautulli listens.";
       };
 
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for Tautulli.";
```
Comment (winterqt) [inline nixos/modules/services/misc/tautulli.nix:38]:
```suggestion
        description = lib.mdDoc "Open ports in the firewall for Tautulli.";
```

**Review by winterqt — CHANGES_REQUESTED** [review]

**Comment by winterqt** [comment]
@ofborg eval

**Review by winterqt — APPROVED** [review]

**Pull request merged**